include/nbl/builtin/hlsl/cpp_compat/concepts.hlsl

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_CPP_COMPAT_CONCEPT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_CPP_COMPAT_CONCEPT_INCLUDED_
```

Comment:
if its meant to be a 1:1 replacement for <concepts> from STL, then don't put it under the `cpp_compat` directory